src/XRP/model/xrp-transaction.ts

```diff
@@ -26,26 +27,59 @@ export default class XrpTransaction {
   public static from(
     getTransactionResponse: GetTransactionResponse,
     xrplNetwork: XrplNetwork,
-  ): XrpTransaction | undefined {
+  ): XrpTransaction {
     const transaction = getTransactionResponse.getTransaction()
     if (!transaction) {
-      return undefined
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Transaction protobuf is missing valid `transaction` field.',
+      )
     }
 
     const account = transaction.getAccount()?.getValue()?.getAddress()
     if (!account) {
-      return undefined
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Transaction protobuf is missing valid `account` field.',
+      )
     }
 
-    const fee = transaction.getFee()?.getDrops()
+    const sourceTag = transaction.getSourceTag()?.getValue()
 
-    const sequence = transaction.getSequence()?.getValue()
+    const sourceXAddress = XrpUtils.encodeXAddress(
+      account,
+      sourceTag,
+      xrplNetwork === XrplNetwork.Test || xrplNetwork === XrplNetwork.Dev,
+    )
+    if (!sourceXAddress) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Cannot construct XAddress from Transaction protobuf `account` and `sourceTag` fields.',
+      )
+    }
 
-    const signingPublicKey = transaction.getSigningPublicKey()?.getValue_asU8()
+    const fee = transaction.getFee()?.getDrops()
+    if (!fee) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Transaction protobuf is missing valid `fee` field.',
+      )
+    }
+    const intFee = Number(fee)
+    if (isNaN(intFee)) {
```

Comment:
is `getDrops` a strong? i think we want to use `BigInteger` here instead cc @amiecorso

Reply:
I was leaving it stored as a `string` for now (`intFee` isn't used at all after this check), just using it as a temp variable to check if `fee` represents an integer value or not.

Reply:
seems reasonable to me - just wanted to make sure there wasn't a "standard" way we validate amounts elsewhere. gogogo

Reply:
Haven't seen this sort of check being applied anywhere, so idk if there's a "standard" way

Reply:
yeah nope there's not